api/app/models/magic_link.py

```diff
@@ -19,22 +19,35 @@ class MagicLink(Base, table=True):
     expiration: datetime = Field(
         default_factory=ten_minutes, description="The expiration date of the magic link"
     )
-    user_uid: UUID = Field(
+    fkey_user_uid: UUID = Field(
         ..., foreign_key="user.uid", description="The user this magic link is for"
     )
 
+    @property
+    def user_id(self) -> str:
+        if uid := self.fkey_user_uid:
+            return f"user-{uid}"
+        return None
+
+    @user_id.setter
+    def user_id(self, value:str) -> None:
+        self.fkey_user_uid = User.to_uid(value)
+
+
     # relationships
-    user: "User" = Relationship(sa_relationship_kwargs={"lazy": "joined"})
+    user: "User" = Relationship(sa_relationship_kwargs={"lazy": "joined", "primaryjoin": "User.uid == MagicLink.fkey_user_uid"})
```

Comment:
So normally sqlalchemy it would be fine to just specify `foreign_keys` array with the column objects, but sqlmodel doesn't convert the columns to Column objects until runtime - so we have to build out the whole primaryjoin string here.